public/pages/DetectorDetail/containers/DetectorDetail.tsx

```diff
@@ -89,13 +91,15 @@ interface DetectorDetailModel {
 export const DetectorDetail = (props: DetectorDetailProps) => {
   const dispatch = useDispatch();
   const detectorId = get(props, 'match.params.detectorId', '') as string;
-  const { detector, hasError } = useFetchDetectorInfo(detectorId);
-  const { monitor, fetchMonitorError } = useFetchMonitorInfo(detectorId);
+  const { detector, hasError, isLoadingDetector } = useFetchDetectorInfo(detectorId);
+  const { monitor, fetchMonitorError, isLoadingMonitor } = useFetchMonitorInfo(detectorId);
+
+  const isLoading = isLoadingDetector || isLoadingMonitor;
```

Comment:
How about we don't show anything is detector is not found or is loading? Show these two tabs may confuse user that they should switch tab to see more.

Reply:
I didn't know you are working on this issue, so I sent out a PR too. You can refer to this PR: https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/pull/156

Reply:
Should consider `hasError` to handle exceptions like `not found` , `timeout` etc. If any problem occur, we already have code to pop up toast:
```
 useEffect(() => {
    if (hasError) {
      toastNotifications.addDanger('Unable to find detector');
      props.history.push('/detectors');
    }
  }, [hasError]);
```

Reply:
> How about we don't show anything if detector not found or still loading? Show these two tabs may confuse user that they should switch tab to see more.

Ok, I agree. Will change.

Reply:
> Should consider `hasError` to handle exceptions like `not found` , `timeout` etc. We already have code to pop up toast and redirect to detector list page.
> 
> ```
>  useEffect(() => {
>     if (hasError) {
>       toastNotifications.addDanger('Unable to find detector');
>       props.history.push('/detectors');
>     }
>   }, [hasError]);
> ```

I agree, I've just tested locally where when I change the URL to an invalid detector id I get in the infinite loading state, which I believe has to do with [this issue](https://github.com/opendistro-for-elasticsearch/anomaly-detection-kibana-plugin/issues/157)

Reply:
Yes, we have some issue for error handling. Let's fix the infinite loading in another PR.